guava/src/com/google/common/escape/ArrayBasedEscaperMap.java

```diff
@@ -65,6 +65,12 @@ private ArrayBasedEscaperMap(char[][] replacementArray) {
   // Creates a replacement array from the given map. The returned array is a
   // linear lookup table of replacement character sequences indexed by the
   // original character value.
+  @SuppressWarnings(value = {"array.length.negative", "array.access.unsafe.low", //char types are non negative: https://github.com/kelloggm/checker-framework/issues/192
+          "array.access.unsafe.high"}
+          /* length of array replacements is maximum value of key values in map
```

Comment:
Why not annotate `map` so that its key set only contains indices for `replacements`, if you're confident that that is true?

You could modify the declaration so that it is `Map<@IndexFor("replacements") char, String>`.

Reply:
I can't use @IndexFor("replacements") since replacements array is declared after map inside the function:
```java
@VisibleForTesting
  static char[][] createReplacementArray(Map<Character, String> map) {
    checkNotNull(map); // GWT specific check (do not optimize)
    if (map.isEmpty()) {
      return EMPTY_REPLACEMENT_ARRAY;
    }
    char max = Collections.max(map.keySet());
    char[][] replacements = new char[max + 1][];
    for (char c : map.keySet()) {
      replacements[c] = map.get(c).toCharArray();
    }
    return replacements;
  }
```